### ISSUE TITLE
High Resolution Image Support (16k+)

### ISSUE THREAD

**Issue by superkitcath** [o3de/o3de#8540]
[Migrated from JIRA ATOM-5559]
Description: Modern GPUs support 16k+ resolution textures so Atom should also

Details: currently the LY AP will fail on images larger then 8k
 - PNG appear to crash (or time out) the builder and only return an error
 **** 8/11/2020 12:51 PM | ERROR | Lost connection to asset builder from Debug
 - If I re-save the image as a JPG it will still fail but return more information (in fact it is timing out, log file attached)

Acceptance Criteria:
 - 16K rgb image will process to compressed texture without failing in the AP
 - Question: Do we need to support even larger texture formats like 32k+?
 - Question: Can we include a project setting for a maximum texture resolution size? If an image is encountered that is larger, it will be downsampled to that target resolution before being compressed - and this should throw a warning!!! (we similarly fix non-power-of-two images, it is better to fall back nicely then completely fail.)

Additional Information:
 - Supporting 16k images should not spawn a debate about whether it's crazy or makes sense, or that users should take a different or better approach.
 - I will check in a Atom project that has 16k images that fail the AP but are not yet uses with assets so nothing actually fails to function but the project will continue to throw warnings and errors

**Comment by Makkkkus** [comment]
Yeah, this is definitely needed for heightmaps for larger areas. Seeing as the terrain can handle insane sizes, this is wouldn't pose any risk.
